paddle/operators/accuracy_op.cu

```diff
@@ -17,21 +17,25 @@ limitations under the License. */
 namespace paddle {
 namespace operators {
 
-__global__ void AccuracySingleKernel(const int N, const int D, const int top_k,
-                                     const int* Xdata, const int* labelData,
-                                     float* accuracy) {
-  int correct = 0;
-  for (int row = 0; row < N; row++) {
-    const int label = labelData[row];
-    for (int col = 0; col < D; col++) {
-      const int pred = Xdata[row * D + col];
-      if (pred == label) {
-        ++correct;
+__global__ void AccuracyCudaKernel(const int N, const int D, const int* Xdata,
+                                   const int* labeldata, float* accuracy) {
+  int count = 0;
+  __shared__ int total;
+  total = 0;
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < (N);
+       i += blockDim.x * gridDim.x) {
+    for (int j = 0; j < D; ++j) {
+      if (Xdata[i * D + j] == labeldata[i]) {
+        ++count;
         break;
       }
     }
   }
-  *accuracy = static_cast<float>(correct) / static_cast<float>(N);
+  atomicAdd(&total, count);
+  __syncthreads();
+  if (threadIdx.x == 0) {
```

Comment:
It seems that when num_samples is greater than 4096 there will be multiple blocks, and this result may be incorrect.

Reply:
Thanks very much for the advise, I'll try to fix it.

Reply:
Done.